olive/engine/engine.py

```diff
@@ -865,23 +890,34 @@ def _cache_run(
         except Exception as e:
             logger.error(f"Failed to cache run: {e}", exc_info=True)
 
-    def _load_run(self, input_model_id: str, pass_name: int, pass_config: dict, accelerator_spec: AcceleratorSpec):
+    def _load_run_json(self, input_model_id: str, pass_name: int, pass_config: dict, accelerator_spec: AcceleratorSpec):
         """
         Load the run from the cache directory.
         """
         input_model_number = input_model_id.split("_")[0]
         run_json_path = self.get_run_json_path(pass_name, input_model_number, pass_config, accelerator_spec)
+        run_json = None
         if run_json_path.exists():
             try:
                 with open(run_json_path, "r") as f:
                     run_json = json.load(f)
-                output_model_id = run_json["output_model_id"]
             except Exception as e:
                 logger.error(f"Failed to load run: {e}", exc_info=True)
-                output_model_id = None
-            return output_model_id
-        else:
-            return None
+                run_json = None
+        return run_json
+
+    def _load_output_model_id_from_run(
```

Comment:
looks like this method is not being used anywhere?

Reply:
Maybe we can combine this with `load_run_json` to return both output_model_id and run_json. if we still want to keep this logic around for future use.

Reply:
Removed directly. Better to keep _load_run clear in my opinion. I changed to make _load_run return empty dict if _load_run failed.